binding/nodejs/src/cobra.ts

```diff
@@ -0,0 +1,181 @@
+};
+
+/**
+ * Node.js binding for Cobra voice activity detection engine
+ *
+ * Performs the calls to the Cobra node library. Does some basic parameter validation to prevent
+ * errors occurring in the library layer. Provides clearer error messages in native JavaScript.
+ */
+export default class Cobra {
+  private _pvCobra: any;
+  private _handle: any;
+
+  private readonly _version: string;
+  private readonly _frameLength: number;
+  private readonly _sampleRate: number;
+
+  /**
+   * Creates an instance of Cobra.
+   * @param {string} accessKey AccessKey obtained from Picovoice Console (https://console.picovoice.ai/).
+   * @param options Optional configuration arguments.
+   * @param {string} options.libraryPath the path to the Cobra library (.node extension)
+   */
+  constructor(accessKey: string, options: CobraOptions = {}) {
+    assert(typeof accessKey === 'string');
+    if (
+      accessKey === null ||
+      accessKey === undefined ||
+      accessKey.length === 0
+    ) {
+      throw new CobraInvalidArgumentError(`No AccessKey provided to Cobra`);
+    }
+
+    const { libraryPath = getSystemLibraryPath() } = options;
+
+    if (!fs.existsSync(libraryPath)) {
+      throw new CobraInvalidArgumentError(
+        `File not found at 'libraryPath': ${libraryPath}`
+      );
+    }
+
+    const pvCobra = require(libraryPath); // eslint-disable-line
+    this._pvCobra = pvCobra;
+
+    let cobraHandleAndStatus: CobraHandleAndStatus | null = null;
+    try {
+      pvCobra.set_sdk('nodejs');
+
+      cobraHandleAndStatus = pvCobra.init(accessKey);
+    } catch (err: any) {
+      pvStatusToException(PvStatus[err.code as keyof typeof PvStatus], err);
+    }
+
+    const status = cobraHandleAndStatus!.status;
+    if (status !== PvStatus.SUCCESS) {
+      this.handlePvStatus(status, 'Cobra failed to initialize');
+    }
+
+    this._handle = cobraHandleAndStatus!.handle;
+    this._sampleRate = pvCobra.sample_rate();
+    this._frameLength = pvCobra.frame_length();
+    this._version = pvCobra.version();
+  }
+
+  /**
+   * @returns number of audio samples per frame (i.e. the length of the array provided to the process function)
+   * @see {@link process}
+   */
+  get frameLength(): number {
+    return this._frameLength;
+  }
+
+  /**
+   * @returns the audio sampling rate accepted by the process function
+   * @see {@link process}
+   */
+  get sampleRate(): number {
+    return this._sampleRate;
+  }
+
+  /**
+   * @returns the version of the Cobra engine
+   */
+  get version(): string {
+    return this._version;
+  }
+
+  /**
+   * Processes a frame of audio and returns the voice probability result.
+   *
+   * @param {Int16Array} pcm Audio data. The audio needs to have a sample rate equal to `Cobra.sampleRate` and be 16-bit linearly-encoded.
+   * The specific array length can be attained by calling `Cobra.frameLength`. This function operates on single-channel audio.
+   * @returns {number} Voice probability result.
```

Comment:
Better to include the range when it's not obvious
```suggestion
   * @returns {number} Probability of voice activity. It is a floating-point number within [0, 1].
```